test/library/standard/Endian/Endian.chpl

```diff
@@ -0,0 +1,228 @@
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /*
+    Support for little-endian and big-endian byte order conversions.
+
+    This module provides a Chapel interface wrapped over the standard
+    C endian module to support little-endian and big-endian byte order
+    conversions. It has functions to convert byte encoding of integer
+    values from the byte order of the host to and from little-endian
+    and big-endian formats.
+ */
+module Endian {
+  use SysCTypes;
+
+  private extern proc htole16(x: c_uint): c_uint;
+  private extern proc le16toh(x: c_uint): c_uint;
+  private extern proc htobe16(x: c_uint): c_uint;
+  private extern proc be16toh(x: c_uint): c_uint;
+  private extern proc htole32(x: c_uint): c_uint;
+  private extern proc le32toh(x: c_uint): c_uint;
+  private extern proc htobe32(x: c_uint): c_uint;
+  private extern proc be32toh(x: c_uint): c_uint;
+  private extern proc htole64(x: c_ulong): c_ulong;
+  private extern proc le64toh(x: c_ulong): c_ulong;
+  private extern proc htobe64(x: c_ulong): c_ulong;
+  private extern proc be64toh(x: c_ulong): c_ulong;
+
+  /*
+    Has value ``true`` if the host system is little-endian type.
+  */
+  const isLittleEndian = hostToLittleEndian(10) == 10;
+
+  /*
+    Has value ``true`` if the host system is big-endian type.
+  */
+  const isBigEndian = hostToBigEndian(10) == 10;
+
+  /* Returns the little endian order of the argument.
+
+      :arg x: The value to convert.
+      :type x: `int`, `uint`
+  */
+  proc hostToLittleEndian(x) {
+    if isIntegralType(x.type) then
+      return _hostToLittleEndian(x);
+    else
+      compilerError("Invalid argument type: expected an uint or int type");
+  }
+
+  /* Returns the host byte order of the argument.
+
+      :arg x: The value to convert.
+      :type x: `int`, `uint`
+  */
+  proc littleEndianToHost(x) {
+    if isIntegralType(x.type) then
+      return _littleEndianToHost(x);
+    else
+      compilerError("Invalid argument type: expected an uint or int type");
+  }
+
+  /* Returns the big endian order of the argument.
+
+      :arg x: The value to convert.
+      :type x: `int`, `uint`
+  */
+  proc hostToBigEndian(x) {
+    if isIntegralType(x.type) then
+      return _hostToBigEndian(x);
+    else
+      compilerError("Invalid argument type: expected an uint or int type");
+  }
+
+  /* Returns the host byte order of the argument.
+
+      :arg x: The value to convert.
+      :type x: `int`, `uint`
+  */
+  proc bigEndianToHost(x) {
+    if isIntegralType(x.type) || isRealType(x.type) then
```

Comment:
Looks like you missed removing `isRealType` here

Reply:
Will wait to do a test run until that's fixed :)

Reply:
Oops! really sorry for that :sweat_smile:

Reply:
Fixed it